src/app/modules/submission/submission.middleware.ts

```diff
@@ -1,14 +1,66 @@
 import { RequestHandler } from 'express'
 import { ParamsDictionary } from 'express-serve-static-core'
+import { StatusCodes } from 'http-status-codes'
+import mongoose from 'mongoose'
 
 import { createLoggerWithLabel } from '../../../config/logger'
 import { WithAutoReplyEmailData } from '../../../types'
+import getSubmissionModel from '../../models/submission.server.model'
 
 import * as SubmissionService from './submission.service'
 import { ProcessedFieldResponse } from './submission.types'
 
+const SubmissionModel = getSubmissionModel(mongoose)
+
 const logger = createLoggerWithLabel(module)
 
+/**
+ * Verifies that by saving this submission, the submission count does not exceed the maximum
+ * specified submissions available (if any).
+ */
+export const checkSubmissionCountBelowMaximum: RequestHandler<
+  { formId: string },
+  unknown,
+  EncryptSubmissionBody
+> = (req, res, next) => {
+  const { form } = req as WithForm<typeof req>
+
+  if (form.submissionLimit !== -1) {
+    return SubmissionModel.countDocuments({ form: form._id })
```

Comment:
The mongoose ODM states that [countDocuments](https://mongoosejs.com/docs/api.html#model_Model.countDocuments) doesn't use indexes, and it seems  like the [native countDocuments](https://docs.mongodb.com/manual/reference/method/db.collection.countDocuments/) is a MongoDB 4.0 feature.

Reply:
I think good old [count](https://docs.mongodb.com/v3.6/reference/method/db.collection.count/#index-use) could work

Reply:
Tested and verified in staging
![image](https://user-images.githubusercontent.com/5690550/107209593-f6bc0a80-6a3d-11eb-8671-b0d584220154.png)

Reply:
So `countDocuments` should work too because it only does a full table scan when the filter is empty. `count` is deprecated, so I was trying to use the new API.

Reply:
I see, thanks! I was looking for guarantees on using the index - seems like the MongoDB team still won't make the query explainable to assuage any concerns: https://jira.mongodb.org/browse/SERVER-41569

Reply:
That's fair, I can use `count` too

Reply:
let's go with `countDocuments`. we have database upgrades on the roadmap, so it would be best if we didn't have to rely on deprecated code